java-frontend/src/main/java/org/sonar/java/SonarComponents.java

```diff
@@ -46,10 +45,8 @@
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
-
-public class SonarComponents implements BatchExtension {
-
-  private static final boolean IS_SONARQUBE_52 = isSonarQube52();
+@BatchSide
+public class SonarComponents {
 
   private final FileLinesContextFactory fileLinesContextFactory;
   private final ResourcePerspectives resourcePerspectives;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "ResourcePerspectives"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)